# Pandas (AI/ML)



In [2]:
import pandas as pd
data = {
    "name":  ["Ananya", "Rohan", "Priya", "Karan", "Simran"],
    "score": [85, 72, 91, 68, 88],
    "grade": ["B", "C", "A", "D", "B"]
}
df = pd.DataFrame(data)
print(df)

     name  score grade
0  Ananya     85     B
1   Rohan     72     C
2   Priya     91     A
3   Karan     68     D
4  Simran     88     B


In [3]:
df

,name,score,grade
0,Ananya,85,B
1,Rohan,72,C
2,Priya,91,A
3,Karan,68,D
4,Simran,88,B


In [4]:
df.head()

,name,score,grade
0,Ananya,85,B
1,Rohan,72,C
2,Priya,91,A
3,Karan,68,D
4,Simran,88,B


In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   name    5 non-null      object
 1   score   5 non-null      int64 
 2   grade   5 non-null      object
dtypes: int64(1), object(2)
memory usage: 252.0+ bytes


In [6]:
df.head(2)

,name,score,grade
0,Ananya,85,B
1,Rohan,72,C


In [7]:
df.tail()

,name,score,grade
0,Ananya,85,B
1,Rohan,72,C
2,Priya,91,A
3,Karan,68,D
4,Simran,88,B


In [8]:
df.tail(3)

,name,score,grade
2,Priya,91,A
3,Karan,68,D
4,Simran,88,B


In [9]:
df.describe()

,score
count,5.00000
mean,80.80000
std,10.18332
min,68.00000
25%,72.00000
50%,85.00000
75%,88.00000
max,91.00000


In [10]:
grade = df["grade"]
grade.head()

,grade
0,B
1,C
2,A
3,D
4,B


In [12]:
print(df.shape)

(5, 3)


In [13]:
print(type(grade))

<class 'pandas.core.series.Series'>


In [16]:
import pandas as pd

data = {
    "name":  ["Ananya", "Rohan", "Priya", "Karan", "Simran"],
    "score": [85, 72, 91, 68, 88],
    "grade": ["B", "C", "A", "D", "B"],
    "revenue": [21231,212245,887997,477885,55579]
}

df = pd.DataFrame(data)
column = df["revenue"]
print(type(column))
print(column.head(3))


<class 'pandas.core.series.Series'>
0     21231
1    212245
2    887997
Name: revenue, dtype: int64


In [17]:
import pandas as pd

age = pd.Series([85,87,69,33,11,55,4,2,8], index = ["Ravi", "Rani", "Raja", "Sena", "Maruthi", "Nani", "Sai", "Manvith", "Naveena"])
print(age)

Ravi       85
Rani       87
Raja       69
Sena       33
Maruthi    11
Nani       55
Sai         4
Manvith     2
Naveena     8
dtype: int64


In [18]:
age.head()

,0
Ravi,85
Rani,87
Raja,69
Sena,33
Maruthi,11


In [19]:
age.info()

<class 'pandas.core.series.Series'>
Index: 9 entries, Ravi to Naveena
Series name: None
Non-Null Count  Dtype
--------------  -----
9 non-null      int64
dtypes: int64(1)
memory usage: 444.0+ bytes


In [20]:
age.describe()

,0
count,9.000000
mean,39.333333
std,35.266840
min,2.000000
25%,8.000000
50%,33.000000
75%,69.000000
max,87.000000


In [21]:
age.tail()

,0
Maruthi,11
Nani,55
Sai,4
Manvith,2
Naveena,8


In [22]:
age.head(5)

,0
Ravi,85
Rani,87
Raja,69
Sena,33
Maruthi,11


In [23]:
print(age.mean())

39.333333333333336


In [25]:
age

,0
Ravi,85
Rani,87
Raja,69
Sena,33
Maruthi,11
Nani,55
Sai,4
Manvith,2
Naveena,8


#Assignment Question

# Task 1

In [3]:
import pandas as pd

df = pd.read_csv("/content/sample_data/orders.csv", index_col="order_id")
df.head()



,customer_name,product,quantity,unit_price,order_date,status
order_id,,,,,,
1001,Alice Johnson,Laptop,1.0,85000.0,2024-01-15,delivered
1002,Bob Smith,Mouse,3.0,499.0,2024-01-16,pending
1003,Carol White,Monitor,2.0,15000.0,NaN,delivered
1004,Dave Brown,Keyboard,NaN,1200.0,2024-01-18,cancelled
1005,Eva Green,Laptop,1.0,85000.0,2024-01-19,delivered


In [4]:
df.tail()

,customer_name,product,quantity,unit_price,order_date,status
order_id,,,,,,
1016,Paul Harris,Mouse,6.0,499.0,2024-01-30,pending
1017,Quinn Martin,Monitor,NaN,14500.0,2024-01-31,delivered
1018,Rachel King,Laptop,1.0,85000.0,NaN,delivered
1019,Sam Wright,Keyboard,3.0,1200.0,2024-02-01,cancelled
1020,Tina Baker,USB Hub,2.0,799.0,2024-02-02,pending


In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 20 entries, 1001 to 1020
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   customer_name  20 non-null     object 
 1   product        20 non-null     object 
 2   quantity       16 non-null     float64
 3   unit_price     20 non-null     float64
 4   order_date     16 non-null     object 
 5   status         20 non-null     object 
dtypes: float64(2), object(4)
memory usage: 1.1+ KB


In [6]:
df.describe()

,quantity,unit_price
count,16.000000,20.000000
mean,2.437500,20224.600000
std,1.504161,33573.389607
min,1.000000,499.000000
25%,1.000000,799.000000
50%,2.000000,2500.000000
75%,3.000000,14625.000000
max,6.000000,85000.000000


In [8]:
df.shape

(20, 6)

In [10]:
df.columns

Index(['customer_name', 'product', 'quantity', 'unit_price', 'order_date',
       'status'],
      dtype='object')

In [13]:
df["order_date"] = pd.to_datetime(df["order_date"])
print(df.dtypes)

customer_name            object
product                  object
quantity                float64
unit_price              float64
order_date       datetime64[ns]
status                   object
dtype: object


# Task 2
1. Rows = 20 and columns = 6
2. quantity and order_date have Null values. Both quantity and order_date have 4 null values each
3. order_date column needs the date-based data type conversion. The function we use to convert --> pd.to_datetime()


# Task 3

In [14]:
unitprice_median = df["unit_price"].median()
print(unitprice_median)

2500.0


In [15]:
df["unit_price"]

,unit_price
order_id,
1001,85000.0
1002,499.0
1003,15000.0
1004,1200.0
1005,85000.0
1006,799.0
1007,2500.0
1008,14500.0
1009,499.0


In [16]:
df["unit_price"].describe()

,unit_price
count,20.000000
mean,20224.600000
std,33573.389607
min,499.000000
25%,799.000000
50%,2500.000000
75%,14625.000000
max,85000.000000


Yes the unit_price column appears to be right skewed. The mean value is greater than the median and the maximum value is far greater than the median.

mean = 20224.6
median = 2500
max = 85000